openwisp_utils/admin_theme/menu.py

```diff
@@ -0,0 +1,154 @@
+from django.apps import registry
+from django.core.exceptions import ImproperlyConfigured
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from ..utils import SortedOrderedDict
+
+MENU = SortedOrderedDict()
+
+
+class BaseMenuItem:
+    def __init__(self, config):
+        if not isinstance(config, dict):
+            raise ImproperlyConfigured(
+                f'"config" should be a type of "dict". Error for config-{config}'
+            )
+
+    def get_context(self, request):
+        return self.create_context(request)
+
+    def create_context(self, request):
+        return {'label': self.label, 'url': self.url, 'icon': self.icon}
+
+
+class MenuItem(BaseMenuItem):
+    def __init__(self, config):
+        super().__init__(config)
+        if config.get('name', None) is not None:
+            if not isinstance(config['name'], str):
+                raise ValueError(
+                    f'"name" of menu item should be a type of "str". Error for config-{config}'
+                )
+            self.name = config['name']
+        else:
+            raise ValueError(f'"name" is missing in the config-{config}')
+        if config.get('model', None) is None:
+            raise ValueError(f'"model" is missing in config-{config}')
+        self.model = config['model']
+        self.label = _(self.get_label(config))
+        self.icon = config.get('icon', None)
+
+    def get_label(self, config=None):
+        if config:
+            if config.get('label', None) is not None:
```

Comment:
Hey, test if  `if config.get('label', None):` gives the same result, `None` is treated like false!